test/test_components.py

```diff
@@ -2955,3 +2956,12 @@ def test_constructor_args():
         "visible": False,
         "value": "Log in please",
     }
+
+
+def test_template_component_configs(io_components):
+    template_components = [c for c in io_components if getattr(c, "is_template", False)]
+    for component in template_components:
+        component_parent_class = inspect.getmro(component)[1]
+        template_config = component().get_config()
+        parent_config = component_parent_class().get_config()
+        assert set(parent_config.keys()).issubset(set(template_config.keys()))
```

Comment:
Why do we assert for subset and not that the keys are identical?

Reply:
Because the custom buttons (gr.LogInButton) are templates but they add some keys. I think checking a subset is ok. We just need to make sure that everything in the parent is represented in the child's config.

Reply:
Ah I didn't realize those were considered templates. Ok sg!